Automattic/woocommerce-payments
# Enable P24 payment method on UPE

Fixes #3181 

#### Changes proposed in this Pull Request


Enable ability to add P24 UPE method for merchants.







#### Testing instructions


* Under dev tools the boxes for _Enable grouped settings_, _Enable UPE checkout_, and _Add UPE additional payment methods_ need to be checked.
* If Polish zloty is an enabled currency in your store, navigate to WooCommerce > Settings > Multi-Currency and remove PLN as a currency.
* Navigate to Payments > Settings.
* Click _Add payment method_, verify P24 is in the modal, along with its icon.
* Select P24, a notice should appear stating that PLN are being added to your store. Click _Add selected_, then scroll to the bottom and click _Save changes_.
* Verify PLN and its logo are now in the _Payment methods_ area.
* Navigate to WooCommerce > Settings > Multi-Currency and verify PLN is now an enabled currency in your store.
* On the frontend, add an item to your cart, and go to checkout.
* Verify your currency *is not* EUR or PLN, payment method should show as _Credit card / debit card_ and when selecting _Use a new payment method_, P24 should not appear.
* Switch your currency to EUR, payment method should show as _Popular payment methods_ and when selecting _Use a new payment method_, P24 should appear.
* Switch your currency to PLN, payment method should show as _Popular payment methods_ and when selecting _Use a new payment method_, P24 should appear.
* Select P24.
* Complete checkout/place order.
* You will be taken to a Stripe test confirmation page, choose _Authorize test payment_.
* Once order is received, navigate to WooCommerce > Orders in the admin.
* Find your order and select it.
* Click _Refund_, and then increase a line item in your order to 1.
* Click _Refund via WooCommerce Payments_, verify your refund completed.
* Navigate to Payments > Transactions, you should see your _Payment_ and _Payment refund_ listed for your order, verify both have the P24 logo.
* Click on the Payment, then verify the _Payment method_ both at the top and bottom are correctly formatting (they will have default test info).
* Do the same for the Payment refund.
* Navigate to Payments > Settings, then click the trash can icon to delete P24. Verify the name and the logo in the modal are correct, then click _Remove_, scroll to the bottom and _Save changes_.
* Verify P24 is no longer in the list of _Payment methods_.


-------------------


**Post merge**
